content/doc/book/pipeline/jenkinsfile.adoc

```diff
@@ -341,6 +341,16 @@ I said, Hello Mr. Jenkins
 Understanding how to use string interpolation is vital for using some of
 Pipeline's more advanced features.
 
+==== [red]*WARNING: Interpolation of Sensitive Variables*
+Groovy string interpolation should [red]*never* be used with sensitive variables. This is
+because the sensitive variable will be interpolated at Groovy evaluation and the variable's
+value could possibly be made available earlier than intended, resulting in the sensitive data leaking.
+
+For example, consider a password variable passed to the shell step. Should Groovy perform the
+interpolation, the password value would become available not only in the shell step, but the
```

Comment:
I would try to be careful to distinguish "variables" and "environment variables" everywhere, and I would give an example of a "bad" Pipeline using `withCredentials` and the `sh` step and then a "good" example showing the right way to do things.

Reply:
Do you think the String Interpolation section should be placed below the environment variables section then? Since we would want to explain that the variables declared in `withCredentials` are environment variables?

Reply:
I don't have a strong opinion on placement. You could link to the environment variables and credentials sections on the page when you mention those topics the first time in the new content.

Reply:
done in 5e8a114acd559b02df299446e548619b622f724a